### DISCUSSION TITLE
Responsive Table Examples?

### DISCUSSION THREAD

**Discussion by rgfx** [bryanmylee/svelte-headless-table#69] [Q&A] [answered]
I noticed there was a grid plugin, but didn't see any examples of how that works.

**Comment by bryanmylee** [comment, marked as answer]
Sorry about that, I've added an example in the docs!

https://svelte-headless-table.bryanmylee.com/docs/plugins/add-grid-layout

The plugin simply swaps out `display: table` for `display: grid`, and provides the correct styles to layout the cells properly.

**Reply by rgfx** [reply]
I see, not very responsive though. Is there a way to make a responsive table by adding data-attr to cell? 

```
<td data-th="Year">1977</td>
```

**Reply by bryanmylee** [reply]
Ah I think I misunderstood what you meant by "responsive".

Correct me if I'm wrong, but you essentially want to add arbitrary data attributes to body cells based on the column and row they are in?

I currently do not have a way to add arbitrary attributes to the cells, but I can either write a quick plugin or extend the `createColumns` function to do so :)

**Reply by rgfx** [reply]
Correct, this one the main ways to make a table responsive. You can see how its done here. https://codepen.io/geoffyuen/pen/DZxEKy. I wonder if this will make column sorting useless.

**Reply by bryanmylee** [reply]
Ah okay, that's the first time I've seen this.

We can definitely support something similar with a plugin to set `display: contents` on the `thead` and `tbody` elements, then add the data attributes accordingly.

**Reply by secretgspot** [reply]
to pile on this discussion.    

I've got access to cell.id which does populate data-title with key of the field. Is it possible to grab the value of (header) instead that is being constructed inside table.createColumns() and put it in?

![image](https://user-images.githubusercontent.com/203323/194728237-cbda878f-2173-4d29-8e58-81e5846a568d.png)
![image](https://user-images.githubusercontent.com/203323/194728258-e810aef8-d99b-434f-8f65-dbb90a916bca.png)
![image](https://user-images.githubusercontent.com/203323/194728311-5d2a57f5-391d-453b-8bab-4109d001e81b.png)

**Reply by bryanmylee** [reply]
@secretgspot The header property can be retrieved with `cell.column.header`. If you're using simple `string` headers, then there should be no issue there.

```svelte
<script>
const table = createTable();
const columns = table.createColumns([
  table.column({
    header: 'First Name', // string returned to `cell.column.header`
    accessor: 'firstName',
  }),
  table.column({
    header: createRender(HeaderCell, {...}), // config returned to `cell.column.header`
    accessor: 'lastName',
  }),
  ...
]);
...
</script>

...
{#each row.cells as cell (cell.id)}
  <Subscribe attrs={cell.attrs()} let:attrs>
    <td {...attrs} data-th={cell.column.header}>
    </td>
  </Subscribe>
{/each}
```

**Reply by rgfx** [reply]
@bryanmylee Thanks this what I was looking for .
